zappa/cli.py

```diff
@@ -474,36 +492,35 @@ def create_package(self):
                 exclude=self.zappa_settings[self.api_stage].get('exclude', [])
             )
 
-        # Throw our setings into it
-        with zipfile.ZipFile(self.zip_path, 'a') as lambda_zip:
-            app_module, app_function = self.app_function.rsplit('.', 1)
-            settings_s = "# Generated by Zappa\nAPP_MODULE='{0!s}'\nAPP_FUNCTION='{1!s}'\n".format(app_module, app_function)
-            
-            if self.debug is not None:
-                settings_s = settings_s + "DEBUG='{0!s}'\n".format((self.debug)) # Cast to Bool in handler
-            settings_s = settings_s + "LOG_LEVEL='{0!s}'\n".format((self.log_level))
-
-            # If we're on a domain, we don't need to define the /<<env>> in
-            # the WSGI PATH
-            if self.domain:
-                settings_s = settings_s + "DOMAIN='{0!s}'\n".format((self.domain))
-            else:
-                settings_s = settings_s + "DOMAIN=None\n"
+        if self.app_function:
+            # Throw custom setings into the zip file
+            with zipfile.ZipFile(self.zip_path, 'a') as lambda_zip:
+                app_module, app_function = self.app_function.rsplit('.', 1)
+                settings_s = "# Generated by Zappa\nAPP_MODULE='{0!s}'\nAPP_FUNCTION='{1!s}'\n".format(app_module, app_function)
 
-            # We can be environment-aware
-            settings_s = settings_s + "API_STAGE='{0!s}'\n".format((self.api_stage))
+                if self.debug:
+                    settings_s = settings_s + "DEBUG='{0!s}'\n".format((self.debug)) # Cast to Bool in handler
+                settings_s = settings_s + "LOG_LEVEL='{0!s}'\n".format((self.log_level))
 
-            # Lambda requires a specific chmod
-            temp_settings = tempfile.NamedTemporaryFile(delete=False)
-            os.chmod(temp_settings.name, 0644)
-            temp_settings.write(settings_s)
-            temp_settings.close()
-            lambda_zip.write(temp_settings.name, 'zappa_settings.py')
-            os.remove(temp_settings.name)
+                # If we're on a domain, we don't need to define the /<<env>> in
+                # the WSGI PATH
+                if self.domain:
+                    settings_s = settings_s + "DOMAIN='{0!s}'\n".format((self.domain))
+                else:
+                    settings_s = settings_s + "DOMAIN=None\n"
 
-            lambda_zip.close()
+                # We can be environment-aware
+                settings_s = settings_s + "API_STAGE='{0!s}'\n".format((self.api_stage))
+
+                # Lambda requires a specific chmod
+                temp_settings = tempfile.NamedTemporaryFile(delete=False)
+                os.chmod(temp_settings.name, 0644)
+                temp_settings.write(settings_s)
+                temp_settings.close()
+                lambda_zip.write(temp_settings.name, 'zappa_settings.py')
+                os.remove(temp_settings.name)
+                # lambda_zip.close()
```

Comment:
probably should just delete this commented line now that this is wrapped in a with zipfile